stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -16,5 +22,76 @@ protected String onGetNodeName(View element) {
         StringUtil.removePrefix(className, "android.view.",
         StringUtil.removePrefix(className, "android.widget."));
   }
+
+  @Override
+  protected void onCopyAttributes(View element, AttributeAccumulator attributes) {
+    String id = getIdAttribute(element);
+    if (id != null) {
+      attributes.add(ID_ATTRIBUTE_NAME, id);
+    }
+  }
+
+  // Copied from Resources.resourceHasPackage because it's marked as @hide
+  private static boolean resourceHasPackage(int resid) {
+    return (resid >>> 24) != 0;
+  }
+
+  @Nullable
+  private static String getIdAttribute(View element) {
+    // Adapted from View.toString()
+
+    int id = element.getId();
+    if (id == View.NO_ID) {
+      return null;
+    }
+
+    String idString = null;
+
+    if (resourceHasPackage(id)) {
+      Resources r = element.getResources();
+      if (r != null) {
+        try {
+          String prefix;
+          String prefixSeparator;
+          switch (id & 0xff000000) {
+            case 0x7f000000:
+              prefix = "";
+              prefixSeparator = "";
+              break;
+            case 0x01000000:
+              prefix = "android";
+              prefixSeparator = ":";
+              break;
+            default:
+              prefix = r.getResourcePackageName(id);
+              prefixSeparator = ":";
+              break;
+          }
+
+          String typeName = r.getResourceTypeName(id);
+          String entryName = r.getResourceEntryName(id);
+
+          // "@" + prefix + prefixSeparator + typeName + "/" + entryName
+          StringBuilder sb = new StringBuilder(1 + prefix.length() + prefixSeparator.length() + typeName.length() + 1 + entryName.length());
+          sb.append("@");
+          sb.append(prefix);
+          sb.append(prefixSeparator);
+          sb.append(typeName);
+          sb.append("/");
+          sb.append(entryName);
+
+          idString = sb.toString();
+
+        } catch (Resources.NotFoundException e) {
+        }
```

Comment:
`LogUtil` -- I don't know what this means in practice; if it ever happens, I want more info :)